augur/utils.py

```diff
@@ -686,3 +686,28 @@ def load_mask_sites(mask_file):
     "A", "G", "C", "T", "U", "N", "R", "Y", "S", "W", "K", "M", "B", "V", "D", "H", "-",
     "a", "g", "c", "t", "u", "n", "r", "y", "s", "w", "k", "m", "b", "v", "d", "h", "-"
 }
+
+
+def read_strains(*files, comment_char="#"):
+    """Reads strain names from one or more plain text files and returns the
+    set of distinct strains.
+
+    Parameters
+    ----------
+    files : one or more str
+        one or more names of text files with one strain name per line
+
+    Returns
+    -------
+    set :
+        strain names from the given input files
+
+    """
+    strains = set()
+    for input_file in files:
+        with open(input_file, 'r', encoding='utf-8') as ifile:
+            for line in ifile:
+                if line[0] != comment_char and len(line.strip()) > 0:
```

Comment:
Is there a format documented somewhere for include/exclude files? I'm sure we used to have examples where lines looked like `<strain> # comment` but looking at the nCoV workflow we don't use this anymore. The code to read exclude files (removed by d3d5bafbdd4d1a18c794f4c46a719f09248a33ea and replaced with this fn) uses `line.split(comment_char)[0].strip()` and i'd recommend adding this in here.

Reply:
Wow, great catch. That is not documented anywhere I can see. That's definitely a bug that include and exclude have had different parsing rules. I'll update to use the more permissive `exclude` version.